web3/eth.py

```diff
@@ -313,3 +314,11 @@ def getCompilers(self):
 
     def getWork(self):
         return self.web3.manager.request_blocking("eth_getWork", [])
+
+    def getGasPrice(self, transaction_params=None):
+        if self.gasPriceStrategy:
+            return self.gasPriceStrategy(self, transaction_params)
+
+    def setGasPriceStrategy(self, gas_price_strategy):
+        # TODO: check passed method has 2 parameters and show handy error
```

Comment:
Given that this is a low level API I think it's ok to skip this check.  Also, it should be allowed for someone to call `web3.eth.setGasPriceStrategy(None)` to remove the strategy unless there is already an alternate way to unset the strategy that I'm not seeing.

Reply:
ok :)  makes sense